### ISSUE TITLE
Questions about interactive decoding 

### ISSUE THREAD

**Issue by summerfan2018** [tensorflow/tensor2tensor#949]
I am trying to translate one sentence at a time using decode_interactively in decoding.py. I am using the already available checkpoints for the English to German translation task (transformer_ende_test) with vocabulary file vocab.ende.32768.

Please could I get clarification on the following concepts:
1. What is extra_length parameter in decode hyperparameters (decode_hp)?
2. What does the parameter num_samples in decode hyperparameters (decode_hp)?
3. What does 'input_modality' mean in decode hyperparameters? Specifically, what is the input vocabulary and output vocabulary?

What should be the values of these parameters when I would want to translate sentences entered by the user (so I have no control over the length of the sentences entered to translate)?

I apologize if these questions are very naive. But it would help if I could get a hint about these concepts.
#

**Comment by martinpopel** [comment]
The length of the translation is restricted to the length of the source sentence plus `extra_length` (which is something like 50 by default, I think). All is counted in subwords.
Theoretically, this could allow translating even very long sentences, but in practice you should (re)segment the user input into reasonably long sentences anyway:
First, long sentences (full paragraphs) may take too much memory and fail with OOM. Second, Transformer does not scale well to sentences longer than seen in the training data, even if extra_length and beam search alpha are set high (at least, this is my experience, in contrast to what is hypothesized in the Attention Is All You Need paper).

I am less sure about the remaining two questions. The input modality in text2text problems (e.g. MT) [should be](https://github.com/tensorflow/tensor2tensor/blob/cb0ae2e960b65278e5891ece38a7f902bbe7548b/tensor2tensor/data_generators/text_problems.py#L300) `registry.Modalities.SYMBOL`, I think.

**Comment by afrozenator** [comment]
Thanks @martinpopel for answering and @summerfan2018 for asking. I'll mark this as closed now, please reopen if you have more questions!

**Comment by ragnarlbrok** [comment]
I have trained a language model on large corpus and trying to decode it in interactive mode with some custom input string but it is getting stopped with following output:



Restoring parameters from lm_model/model.ckpt.2708000
INFO:tensorflow:Running local_init_op.
INFO:Done running local_init_op.
INFO:tensorflow:



I have debugged the code and found that it ran fine till before the 'estimator.predict(input_fn, checkpoint_path=checkpoint _path)' line in decode_interactively( ) function of decoding.py file but after that it did not produced any output of predict function and did not entered into for loop in last of decode_interactively( ) function.
